[Bug]: Milvus standalone and rootcoord pod  will restart when init an object of MilvusSys class

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: master-20211122-7a0d5ed
- Deployment mode(standalone or cluster): standalone and cluster
- SDK version(e.g. pymilvus v2.0.0rc2):2.0.0rc9.dev13
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```


### Current Behavior

There are some error messages and Milvus will restart.
```
[2021-11-23 10:27:14 - DEBUG - ci_test]: (api_response) : (insert count: 10, delete count: 0, upsert count: 0, timestamp: 429296041141469193)  (api_request.py:21)
[2021-11-23 10:27:14 - ERROR - pymilvus.client.grpc_handler]:
Addr [10.96.185.38:19530] query
RPC error: <_MultiThreadedRendezvous of RPC that terminated with:
	status = StatusCode.UNAVAILABLE
	details = "Socket closed"
	debug_error_string = "{"created":"@1637634434.287467000","description":"Error received from peer ipv4:127.0.0.1:7890","file":"src/core/lib/surface/call.cc","file_line":1068,"grpc_message":"Socket closed","grpc_status":14}"
>
	{'API start': '2021-11-23 10:27:14.205622', 'RPC start': '2021-11-23 10:27:14.212338', 'RPC error': '2021-11-23 10:27:14.288097'} (grpc_handler.py:83)
[2021-11-23 10:27:14 - ERROR - pymilvus.client.grpc_handler]:
Addr [10.96.185.38:19530] list_partitions
RPC error: <_MultiThreadedRendezvous of RPC that terminated with:
	status = StatusCode.UNAVAILABLE
	details = "Socket closed"
	debug_error_string = "{"created":"@1637634434.287666000","description":"Error received from peer ipv4:127.0.0.1:7890","file":"src/core/lib/surface/call.cc","file_line":1068,"grpc_message":"Socket closed","grpc_status":14}"
>
	{'API start': '2021-11-23 10:27:14.249368', 'RPC start': '2021-11-23 10:27:14.249971', 'RPC error': '2021-11-23 10:27:14.289062'} (grpc_handler.py:83)
[2021-11-23 10:27:14 - ERROR - pymilvus.client.grpc_handler]:
Addr [10.96.185.38:19530] flush
RPC error: <_MultiThreadedRendezvous of RPC that terminated with:
	status = StatusCode.UNAVAILABLE
	details = "Socket closed"
	debug_error_string = "{"created":"@1637634434.287626000","description":"Error received from peer ipv4:127.0.0.1:7890","file":"src/core/lib/surface/call.cc","file_line":1068,"grpc_message":"Socket closed","grpc_status":14}"
>
	{'API start': '2021-11-23 10:27:14.248468', 'RPC start': '2021-11-23 10:27:14.249172', 'RPC error': '2021-11-23 10:27:14.289846'} (grpc_handler.py:83)
```


### Expected Behavior

works well

### Steps To Reproduce
1. cd `tests/python_client/chaos`

2. modify param value and run script `chaos_test.sh`
pod="standalone" # if test for rootcoord, replace standalone to querynode or etcd
chaos_type="pod_kill"
chaos_task="chaos-test"
3. run `bash chaos_test.sh`
### Anything else?
This is not about chaos, because the error happens before chaos injection

[standalone-restart-11-23.zip](https://github.com/milvus-io/milvus/files/7585620/standalone-restart-11-23.zip)

[rootcoord-restart-11-23-15-47.zip](https://github.com/milvus-io/milvus/files/7586458/rootcoord-restart-11-23-15-47.zip)

There is another side effect of this issue: after restarting of rootcoord, collections.num_entities  (flush ) will raise error
```
self = <_MultiThreadedRendezvous of RPC that terminated with:
	status = StatusCode.UNIMPLEMENTED
	details = "unknown method G...":1068,"grpc_message":"unknown method GetFlushState for service milvus.proto.milvus.MilvusService","grpc_status":12}"
>
timeout = None

    def result(self, timeout=None):
        """Returns the result of the computation or raises its exception.

        See grpc.Future.result for the full API contract.
        """
        with self._state.condition:
            timed_out = _common.wait(self._state.condition.wait,
                                     self._is_complete,
                                     timeout=timeout)
            if timed_out:
                raise grpc.FutureTimeoutError()
            else:
                if self._state.code is grpc.StatusCode.OK:
                    return self._state.response
                elif self._state.cancelled:
                    raise grpc.FutureCancelledError()
                else:
>                   raise self
E                   grpc._channel._MultiThreadedRendezvous: <_MultiThreadedRendezvous of RPC that terminated with:
E                   	status = StatusCode.UNIMPLEMENTED
E                   	details = "unknown method GetFlushState for service milvus.proto.milvus.MilvusService"
E                   	debug_error_string = "{"created":"@1637671483.562784000","description":"Error received from peer ipv4:127.0.0.1:7890","file":"src/core/lib/surface/call.cc","file_line":1068,"grpc_message":"unknown method GetFlushState for service milvus.proto.milvus.MilvusService","grpc_status":12}"
E                   >
```

--- a/cmd/main.go
+++ b/cmd/main.go
@@ -72,9 +72,29 @@ func printBanner() {
 func injectVariablesToEnv() {
 	// inject in need
 
-	err := os.Setenv(metricsinfo.GitCommitEnvKey, GitCommit)
+	var err error
+
+	err = os.Setenv(metricsinfo.GitCommitEnvKey, GitCommit)
 	if err != nil {
-		log.Warn("failed to inject git commit to environment variable",
+		log.Warn(fmt.Sprintf("failed to inject %s to environment variable", metricsinfo.GitCommitEnvKey),
+			zap.Error(err))
+	}
+
+	err = os.Setenv(metricsinfo.GitBuildTagsEnvKey, BuildTags)
+	if err != nil {
+		log.Warn(fmt.Sprintf("failed to inject %s to environment variable", metricsinfo.GitBuildTagsEnvKey),
+			zap.Error(err))
+	}
+
+	err = os.Setenv(metricsinfo.MilvusBuildTimeEnvKey, BuildTime)
+	if err != nil {
+		log.Warn(fmt.Sprintf("failed to inject %s to environment variable", metricsinfo.MilvusBuildTimeEnvKey),
+			zap.Error(err))
+	}
+
+	err = os.Setenv(metricsinfo.MilvusUsedGoVersion, GoVersion)
+	if err != nil {
+		log.Warn(fmt.Sprintf("failed to inject %s to environment variable", metricsinfo.MilvusUsedGoVersion),
 			zap.Error(err))
 	}
 }

--- a/internal/datacoord/metrics_info.go
+++ b/internal/datacoord/metrics_info.go
@@ -19,7 +19,6 @@ package datacoord
 import (
 	"context"
 	"errors"
-	"os"
 
 	"github.com/milvus-io/milvus/internal/util/uniquegenerator"
 
@@ -90,7 +89,7 @@ func (s *Server) getSystemInfoMetrics(
 
 // getDataCoordMetrics composes datacoord infos
 func (s *Server) getDataCoordMetrics() metricsinfo.DataCoordInfos {
-	return metricsinfo.DataCoordInfos{
+	ret := metricsinfo.DataCoordInfos{
 		BaseComponentInfos: metricsinfo.BaseComponentInfos{
 			Name: metricsinfo.ConstructComponentName(typeutil.DataCoordRole, Params.NodeID),
 			HardwareInfos: metricsinfo.HardwareMetrics{
@@ -102,10 +101,7 @@ func (s *Server) getDataCoordMetrics() metricsinfo.DataCoordInfos {
 				Disk:         metricsinfo.GetDiskCount(),
 				DiskUsage:    metricsinfo.GetDiskUsage(),
 			},
-			SystemInfo: metricsinfo.DeployMetrics{
-				SystemVersion: os.Getenv(metricsinfo.GitCommitEnvKey),
-				DeployMode:    os.Getenv(metricsinfo.DeployModeEnvKey),
-			},
+			SystemInfo:  metricsinfo.DeployMetrics{},
 			CreatedTime: Params.CreatedTime.String(),
 			UpdatedTime: Params.UpdatedTime.String(),
 			Type:        typeutil.DataCoordRole,
@@ -115,6 +111,10 @@ func (s *Server) getDataCoordMetrics() metricsinfo.DataCoordInfos {
 			SegmentMaxSize: Params.SegmentMaxSize,
 		},
 	}
+
+	metricsinfo.FillDeployMetricsWithEnv(&ret.BaseComponentInfos.SystemInfo)
+
+	return ret
 }
 
 // getDataNodeMetrics composes data node infos

--- a/internal/datanode/metrics_info.go
+++ b/internal/datanode/metrics_info.go
@@ -18,7 +18,6 @@ package datanode
 
 import (
 	"context"
-	"os"
 
 	"github.com/milvus-io/milvus/internal/proto/commonpb"
 	"github.com/milvus-io/milvus/internal/proto/milvuspb"
@@ -40,10 +39,7 @@ func (node *DataNode) getSystemInfoMetrics(ctx context.Context, req *milvuspb.Ge
 				Disk:         metricsinfo.GetDiskCount(),
 				DiskUsage:    metricsinfo.GetDiskUsage(),
 			},
-			SystemInfo: metricsinfo.DeployMetrics{
-				SystemVersion: os.Getenv(metricsinfo.GitCommitEnvKey),
-				DeployMode:    os.Getenv(metricsinfo.DeployModeEnvKey),
-			},
+			SystemInfo:  metricsinfo.DeployMetrics{},
 			CreatedTime: Params.CreatedTime.String(),
 			UpdatedTime: Params.UpdatedTime.String(),
 			Type:        typeutil.DataNodeRole,
@@ -53,6 +49,9 @@ func (node *DataNode) getSystemInfoMetrics(ctx context.Context, req *milvuspb.Ge
 			FlushInsertBufferSize: Params.FlushInsertBufferSize,
 		},
 	}
+
+	metricsinfo.FillDeployMetricsWithEnv(&nodeInfos.SystemInfo)
+
 	resp, err := metricsinfo.MarshalComponentInfos(nodeInfos)
 	if err != nil {
 		return &milvuspb.GetMetricsResponse{

--- a/internal/indexcoord/metrics_info.go
+++ b/internal/indexcoord/metrics_info.go
@@ -18,7 +18,6 @@ package indexcoord
 
 import (
 	"context"
-	"os"
 
 	"github.com/milvus-io/milvus/internal/util/uniquegenerator"
 
@@ -53,10 +52,7 @@ func getSystemInfoMetrics(
 					Disk:         metricsinfo.GetDiskCount(),
 					DiskUsage:    metricsinfo.GetDiskUsage(),
 				},
-				SystemInfo: metricsinfo.DeployMetrics{
-					SystemVersion: os.Getenv(metricsinfo.GitCommitEnvKey),
-					DeployMode:    os.Getenv(metricsinfo.DeployModeEnvKey),
-				},
+				SystemInfo:  metricsinfo.DeployMetrics{},
 				CreatedTime: Params.CreatedTime.String(),
 				UpdatedTime: Params.UpdatedTime.String(),
 				Type:        typeutil.IndexCoordRole,
@@ -68,6 +64,8 @@ func getSystemInfoMetrics(
 		},
 		ConnectedNodes: make([]metricsinfo.IndexNodeInfos, 0),
 	}
+
+	metricsinfo.FillDeployMetricsWithEnv(&clusterTopology.Self.SystemInfo)
 
 	nodesMetrics := coord.nodeManager.getMetrics(ctx, req)
 	for _, nodeMetrics := range nodesMetrics {

--- a/internal/indexnode/metrics_info.go
+++ b/internal/indexnode/metrics_info.go
@@ -18,7 +18,6 @@ package indexnode
 
 import (
 	"context"
-	"os"
 
 	"github.com/milvus-io/milvus/internal/proto/commonpb"
 	"github.com/milvus-io/milvus/internal/util/metricsinfo"
@@ -46,10 +45,7 @@ func getSystemInfoMetrics(
 				Disk:         metricsinfo.GetDiskCount(),
 				DiskUsage:    metricsinfo.GetDiskUsage(),
 			},
-			SystemInfo: metricsinfo.DeployMetrics{
-				SystemVersion: os.Getenv(metricsinfo.GitCommitEnvKey),
-				DeployMode:    os.Getenv(metricsinfo.DeployModeEnvKey),
-			},
+			SystemInfo:  metricsinfo.DeployMetrics{},
 			CreatedTime: Params.CreatedTime.String(),
 			UpdatedTime: Params.UpdatedTime.String(),
 			Type:        typeutil.IndexNodeRole,
@@ -61,6 +57,9 @@ func getSystemInfoMetrics(
 			SimdType: Params.SimdType,
 		},
 	}
+
+	metricsinfo.FillDeployMetricsWithEnv(&nodeInfos.SystemInfo)
+
 	resp, err := metricsinfo.MarshalComponentInfos(nodeInfos)
 	if err != nil {
 		return &milvuspb.GetMetricsResponse{

--- a/internal/proxy/impl.go
+++ b/internal/proxy/impl.go
@@ -2422,6 +2422,19 @@ func (node *Proxy) GetMetrics(ctx context.Context, req *milvuspb.GetMetricsReque
 	log.Debug("Proxy.GetMetrics",
 		zap.String("metric_type", metricType))
 
+	msgID := UniqueID(0)
+	msgID, err = node.idAllocator.AllocOne()
+	if err != nil {
+		log.Warn("Proxy.GetMetrics failed to allocate id",
+			zap.Error(err))
+	}
+	req.Base = &commonpb.MsgBase{
+		MsgType:   commonpb.MsgType_SystemInfo,
+		MsgID:     msgID,
+		Timestamp: 0,
+		SourceID:  Params.ProxyID,
+	}
+
 	if metricType == metricsinfo.SystemInfoMetrics {
 		ret, err := node.metricsCacheManager.GetSystemInfoMetrics()
 		if err == nil && ret != nil {

--- a/internal/proxy/metrics_info.go
+++ b/internal/proxy/metrics_info.go
@@ -18,7 +18,6 @@ package proxy
 
 import (
 	"context"
-	"os"
 
 	"github.com/milvus-io/milvus/internal/proto/commonpb"
 
@@ -48,6 +47,7 @@ func getSystemInfoMetrics(
 
 	proxyRoleName := metricsinfo.ConstructComponentName(typeutil.ProxyRole, Params.ProxyID)
 	identifierMap[proxyRoleName] = int(node.session.ServerID)
+
 	proxyTopologyNode := metricsinfo.SystemTopologyNode{
 		Identifier: int(node.session.ServerID),
 		Connected:  make([]metricsinfo.ConnectionEdge, 0),
@@ -65,10 +65,7 @@ func getSystemInfoMetrics(
 					Disk:         metricsinfo.GetDiskCount(),
 					DiskUsage:    metricsinfo.GetDiskUsage(),
 				},
-				SystemInfo: metricsinfo.DeployMetrics{
-					SystemVersion: os.Getenv(metricsinfo.GitCommitEnvKey),
-					DeployMode:    os.Getenv(metricsinfo.DeployModeEnvKey),
-				},
+				SystemInfo:  metricsinfo.DeployMetrics{},
 				CreatedTime: Params.CreatedTime.String(),
 				UpdatedTime: Params.UpdatedTime.String(),
 				Type:        typeutil.ProxyRole,
@@ -80,6 +77,7 @@ func getSystemInfoMetrics(
 			},
 		},
 	}
+	metricsinfo.FillDeployMetricsWithEnv(&(proxyTopologyNode.Infos.(*metricsinfo.ProxyInfos).SystemInfo))
 
 	queryCoordResp, queryCoordErr := node.queryCoord.GetMetrics(ctx, request)
 	queryCoordRoleName := ""

--- a/internal/querycoord/metrics_info.go
+++ b/internal/querycoord/metrics_info.go
@@ -18,7 +18,6 @@ package querycoord
 
 import (
 	"context"
-	"os"
 
 	"github.com/milvus-io/milvus/internal/util/uniquegenerator"
 
@@ -53,10 +52,7 @@ func getSystemInfoMetrics(
 					Disk:         metricsinfo.GetDiskCount(),
 					DiskUsage:    metricsinfo.GetDiskUsage(),
 				},
-				SystemInfo: metricsinfo.DeployMetrics{
-					SystemVersion: os.Getenv(metricsinfo.GitCommitEnvKey),
-					DeployMode:    os.Getenv(metricsinfo.DeployModeEnvKey),
-				},
+				SystemInfo:  metricsinfo.DeployMetrics{},
 				CreatedTime: Params.CreatedTime.String(),
 				UpdatedTime: Params.UpdatedTime.String(),
 				Type:        typeutil.QueryCoordRole,
@@ -69,6 +65,7 @@ func getSystemInfoMetrics(
 		},
 		ConnectedNodes: make([]metricsinfo.QueryNodeInfos, 0),
 	}
+	metricsinfo.FillDeployMetricsWithEnv(&clusterTopology.Self.SystemInfo)
 
 	nodesMetrics := qc.cluster.getMetrics(ctx, req)
 	for _, nodeMetrics := range nodesMetrics {

--- a/internal/querynode/metrics_info.go
+++ b/internal/querynode/metrics_info.go
@@ -13,7 +13,6 @@ package querynode
 
 import (
 	"context"
-	"os"
 
 	"github.com/milvus-io/milvus/internal/proto/commonpb"
 	"github.com/milvus-io/milvus/internal/proto/milvuspb"
@@ -42,10 +41,7 @@ func getSystemInfoMetrics(ctx context.Context, req *milvuspb.GetMetricsRequest, 
 				Disk:         metricsinfo.GetDiskCount(),
 				DiskUsage:    metricsinfo.GetDiskUsage(),
 			},
-			SystemInfo: metricsinfo.DeployMetrics{
-				SystemVersion: os.Getenv(metricsinfo.GitCommitEnvKey),
-				DeployMode:    os.Getenv(metricsinfo.DeployModeEnvKey),
-			},
+			SystemInfo:  metricsinfo.DeployMetrics{},
 			CreatedTime: Params.CreatedTime.String(),
 			UpdatedTime: Params.UpdatedTime.String(),
 			Type:        typeutil.QueryNodeRole,
@@ -62,6 +58,8 @@ func getSystemInfoMetrics(ctx context.Context, req *milvuspb.GetMetricsRequest, 
 			SimdType: Params.SimdType,
 		},
 	}
+	metricsinfo.FillDeployMetricsWithEnv(&nodeInfos.SystemInfo)
+
 	resp, err := metricsinfo.MarshalComponentInfos(nodeInfos)
 	if err != nil {
 		return &milvuspb.GetMetricsResponse{

--- a/internal/rootcoord/metrics_info.go
+++ b/internal/rootcoord/metrics_info.go
@@ -13,7 +13,6 @@ package rootcoord
 
 import (
 	"context"
-	"os"
 
 	"go.uber.org/zap"
 
@@ -39,10 +38,7 @@ func (c *Core) getSystemInfoMetrics(ctx context.Context, req *milvuspb.GetMetric
 					Disk:         metricsinfo.GetDiskCount(),
 					DiskUsage:    metricsinfo.GetDiskUsage(),
 				},
-				SystemInfo: metricsinfo.DeployMetrics{
-					SystemVersion: os.Getenv(metricsinfo.GitCommitEnvKey),
-					DeployMode:    os.Getenv(metricsinfo.DeployModeEnvKey),
-				},
+				SystemInfo:  metricsinfo.DeployMetrics{},
 				CreatedTime: Params.CreatedTime.String(),
 				UpdatedTime: Params.UpdatedTime.String(),
 				Type:        typeutil.RootCoordRole,
@@ -58,6 +54,7 @@ func (c *Core) getSystemInfoMetrics(ctx context.Context, req *milvuspb.GetMetric
 			ConnectedComponents: []metricsinfo.ConnectionInfo{},
 		},
 	}
+	metricsinfo.FillDeployMetricsWithEnv(&rootCoordTopology.Self.SystemInfo)
 
 	resp, err := metricsinfo.MarshalTopology(rootCoordTopology)
 	if err != nil {

--- a/internal/util/metricsinfo/metrics_info.go
+++ b/internal/util/metricsinfo/metrics_info.go
@@ -57,12 +57,24 @@ const (
 
 	// StandaloneDeployMode represents the stand-alone deployment mode
 	StandaloneDeployMode = "STANDALONE"
+
+	// GitBuildTagsEnvKey build tag
+	GitBuildTagsEnvKey = "MILVUS_GIT_BUILD_TAGS"
+
+	// MilvusBuildTimeEnvKey build time
+	MilvusBuildTimeEnvKey = "MILVUS_BUILD_TIME"
+
+	// MilvusUsedGoVersion used go version
+	MilvusUsedGoVersion = "MILVUS_USED_GO_VERSION"
 )
 
 // DeployMetrics records the deploy information of nodes.
 type DeployMetrics struct {
 	SystemVersion string `json:"system_version"`
 	DeployMode    string `json:"deploy_mode"`
+	BuildVersion  string `json:"build_version"`
+	BuildTime     string `json:"build_time"`
+	UsedGoVersion string `json:"used_go_version"`
 }
 
 // BaseComponentInfos contains basic information that all components should have.

--- a/internal/util/metricsinfo/metrics_info_test.go
+++ b/internal/util/metricsinfo/metrics_info_test.go
@@ -40,6 +40,9 @@ func TestBaseComponentInfos_Codec(t *testing.T) {
 		SystemInfo: DeployMetrics{
 			SystemVersion: "8b1ae98fa97ce1c7ba853e8b9ff1c7ce24458dc1",
 			DeployMode:    ClusterDeployMode,
+			BuildVersion:  "2.0.0-rc8",
+			BuildTime:     "2021-11-24, 11:37:25",
+			UsedGoVersion: "go version go1.15.2 linux/amd64",
 		},
 		CreatedTime: time.Now().String(),
 		UpdatedTime: time.Now().String(),
@@ -74,6 +77,9 @@ func TestQueryNodeInfos_Codec(t *testing.T) {
 			SystemInfo: DeployMetrics{
 				SystemVersion: "8b1ae98fa97ce1c7ba853e8b9ff1c7ce24458dc1",
 				DeployMode:    ClusterDeployMode,
+				BuildVersion:  "2.0.0-rc8",
+				BuildTime:     "2021-11-24, 11:37:25",
+				UsedGoVersion: "go version go1.15.2 linux/amd64",
 			},
 			CreatedTime: time.Now().String(),
 			UpdatedTime: time.Now().String(),
@@ -119,6 +125,9 @@ func TestQueryCoordInfos_Codec(t *testing.T) {
 			SystemInfo: DeployMetrics{
 				SystemVersion: "8b1ae98fa97ce1c7ba853e8b9ff1c7ce24458dc1",
 				DeployMode:    ClusterDeployMode,
+				BuildVersion:  "2.0.0-rc8",
+				BuildTime:     "2021-11-24, 11:37:25",
+				UsedGoVersion: "go version go1.15.2 linux/amd64",
 			},
 			CreatedTime: time.Now().String(),
 			UpdatedTime: time.Now().String(),
@@ -158,6 +167,9 @@ func TestIndexNodeInfos_Codec(t *testing.T) {
 			SystemInfo: DeployMetrics{
 				SystemVersion: "8b1ae98fa97ce1c7ba853e8b9ff1c7ce24458dc1",
 				DeployMode:    ClusterDeployMode,
+				BuildVersion:  "2.0.0-rc8",
+				BuildTime:     "2021-11-24, 11:37:25",
+				UsedGoVersion: "go version go1.15.2 linux/amd64",
 			},
 			CreatedTime: time.Now().String(),
 			UpdatedTime: time.Now().String(),
@@ -198,6 +210,9 @@ func TestIndexCoordInfos_Codec(t *testing.T) {
 			SystemInfo: DeployMetrics{
 				SystemVersion: "8b1ae98fa97ce1c7ba853e8b9ff1c7ce24458dc1",
 				DeployMode:    ClusterDeployMode,
+				BuildVersion:  "2.0.0-rc8",
+				BuildTime:     "2021-11-24, 11:37:25",
+				UsedGoVersion: "go version go1.15.2 linux/amd64",
 			},
 			CreatedTime: time.Now().String(),
 			UpdatedTime: time.Now().String(),
@@ -236,6 +251,9 @@ func TestDataNodeInfos_Codec(t *testing.T) {
 			SystemInfo: DeployMetrics{
 				SystemVersion: "8b1ae98fa97ce1c7ba853e8b9ff1c7ce24458dc1",
 				DeployMode:    ClusterDeployMode,
+				BuildVersion:  "2.0.0-rc8",
+				BuildTime:     "2021-11-24, 11:37:25",
+				UsedGoVersion: "go version go1.15.2 linux/amd64",
 			},
 			CreatedTime: time.Now().String(),
 			UpdatedTime: time.Now().String(),
@@ -274,6 +292,9 @@ func TestDataCoordInfos_Codec(t *testing.T) {
 			SystemInfo: DeployMetrics{
 				SystemVersion: "8b1ae98fa97ce1c7ba853e8b9ff1c7ce24458dc1",
 				DeployMode:    ClusterDeployMode,
+				BuildVersion:  "2.0.0-rc8",
+				BuildTime:     "2021-11-24, 11:37:25",
+				UsedGoVersion: "go version go1.15.2 linux/amd64",
 			},
 			CreatedTime: time.Now().String(),
 			UpdatedTime: time.Now().String(),
@@ -312,6 +333,9 @@ func TestRootCoordInfos_Codec(t *testing.T) {
 			SystemInfo: DeployMetrics{
 				SystemVersion: "8b1ae98fa97ce1c7ba853e8b9ff1c7ce24458dc1",
 				DeployMode:    ClusterDeployMode,
+				BuildVersion:  "2.0.0-rc8",
+				BuildTime:     "2021-11-24, 11:37:25",
+				UsedGoVersion: "go version go1.15.2 linux/amd64",
 			},
 			CreatedTime: time.Now().String(),
 			UpdatedTime: time.Now().String(),

--- a/internal/util/metricsinfo/utils.go
+++ b/internal/util/metricsinfo/utils.go
@@ -1,0 +1,25 @@
+// Copyright (C) 2019-2020 Zilliz. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions and limitations under the License.
+
+package metricsinfo
+
+import (
+	"os"
+)
+
+// FillDeployMetricsWithEnv fill deploy metrics with env.
+func FillDeployMetricsWithEnv(m *DeployMetrics) {
+	m.SystemVersion = os.Getenv(GitCommitEnvKey)
+	m.DeployMode = os.Getenv(DeployModeEnvKey)
+	m.BuildVersion = os.Getenv(GitBuildTagsEnvKey)
+	m.UsedGoVersion = os.Getenv(MilvusUsedGoVersion)
+	m.BuildTime = os.Getenv(MilvusBuildTimeEnvKey)
+}

--- a/internal/util/metricsinfo/utils_test.go
+++ b/internal/util/metricsinfo/utils_test.go
@@ -1,0 +1,73 @@
+// Copyright (C) 2019-2020 Zilliz. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions and limitations under the License.
+
+package metricsinfo
+
+import (
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestFillDeployMetricsWithEnv(t *testing.T) {
+	var err error
+
+	var m DeployMetrics
+
+	commit := "commit"
+	originalCommit := os.Getenv(GitCommitEnvKey)
+	err = os.Setenv(GitCommitEnvKey, commit)
+	assert.NoError(t, err)
+
+	deploy := "deploy"
+	originalDeploy := os.Getenv(DeployModeEnvKey)
+	err = os.Setenv(DeployModeEnvKey, deploy)
+	assert.NoError(t, err)
+
+	version := "version"
+	originalVersion := os.Getenv(GitBuildTagsEnvKey)
+	err = os.Setenv(GitBuildTagsEnvKey, version)
+	assert.NoError(t, err)
+
+	goVersion := "go"
+	originalGoVersion := os.Getenv(MilvusUsedGoVersion)
+	err = os.Setenv(MilvusUsedGoVersion, goVersion)
+	assert.NoError(t, err)
+
+	buildTime := "build"
+	originalBuildTime := os.Getenv(MilvusBuildTimeEnvKey)
+	err = os.Setenv(MilvusBuildTimeEnvKey, buildTime)
+	assert.NoError(t, err)
+
+	FillDeployMetricsWithEnv(&m)
+	assert.NotNil(t, m)
+	assert.Equal(t, commit, m.SystemVersion)
+	assert.Equal(t, deploy, m.DeployMode)
+	assert.Equal(t, version, m.BuildVersion)
+	assert.Equal(t, goVersion, m.UsedGoVersion)
+	assert.Equal(t, buildTime, m.BuildTime)
+
+	err = os.Setenv(GitCommitEnvKey, originalCommit)
+	assert.NoError(t, err)
+
+	err = os.Setenv(DeployModeEnvKey, originalDeploy)
+	assert.NoError(t, err)
+
+	err = os.Setenv(GitBuildTagsEnvKey, originalVersion)
+	assert.NoError(t, err)
+
+	err = os.Setenv(MilvusUsedGoVersion, originalGoVersion)
+	assert.NoError(t, err)
+
+	err = os.Setenv(MilvusBuildTimeEnvKey, originalBuildTime)
+	assert.NoError(t, err)
+}

